### ISSUE TITLE
Define π for CCE rollouts: greedy target policy vs ε-greedy behaviour policy — and why it decides whether n=1 is exact

### ISSUE THREAD

**Issue by manchado53** [manchado53/counterfactual-reasoning#23]
The paper never pins down what **π** is in Algorithm 1, and the three places it is
described disagree. That is not pedantry: the choice decides whether
`cf_n_rollouts=1` is an *exact* estimate or an under-sampled one, so it changes the
cheapest configuration we have.

Related but distinct from #20 (which is about the temperature knob and Jeremy's
compute argument) and #15 (which is about the scoring gate).

## The three definitions, verbatim

```
 paper.tex:128   "π(a|s), the probability that a TRAINED agent should take action a"
 paper.tex:97    "rolls out n counterfactual trajectories ... using the CURRENT policy"
 paper.tex:145   "the agent took an historic action a_t according to π(s_t)"
```

Line 145 is the binding one — it says the actions **stored in the buffer** were drawn
from π. In our code those come from the **ε-greedy behaviour policy** (ε: 1.0 → 0.05).
So a literal reading makes π = ε-greedy, and *neither* rollout setting we have ever run
is π:

| setting | what it is | is it π (=ε-greedy)? |
|---|---|---|
| `cf_rollout_temperature=0.5` (shipped, all JaxNav sweeps to date) | softmax(Q/0.5), follows argmax ~35% of the time, temperature FIXED while ε decays | no, at any point in training |
| `cf_rollout_temperature=0` (running now, array 275427) | argmax = the greedy/target policy | no, but within 5% of π after ε decays |
| ε-greedy at the current ε | π exactly | **never implemented** |

## Why it is not cosmetic: it decides whether n=1 is exact

```
 π = GREEDY (deterministic)
     rollout is deterministic in a deterministic env
     -> d_G(s,a) is a point mass
     -> n = 1 estimates it EXACTLY

 π = ε-GREEDY (stochastic, ε=0.05)
     rollout is stochastic even in a deterministic env
     -> d_G(s,a) has real spread
     -> n = 1 is a one-sample estimate of a distribution
```

Our whole coverage gain (0.86% → 16.05% measured, array 275427) is funded by
`cf_n_rollouts=1`. Under the ε-greedy reading that funding is not free.

## Measured: the imagined world does not match the real one

101 instrumented runs, `rollout_*_frac` at the last eval against `eval.jsonl` over the
final 20 evals:

```
 outcome        rollout (T=0.5, H=60)    eval (argmax, full episode)
 goal                    9.7%                    18.8%
 crash                  46.7%                    23.3%
 timeout                 7.3%                    57.9%
 unresolved             36.4%                      --
```

The rollout reaches the goal **half** as often and crashes **twice** as often as the
policy actually being evaluated. Caveat, stated plainly: the horizon also differs
(H=60 vs full 200-step episodes), and the 36.4% unresolved would mostly resolve to
timeouts, so the timeout row is not a clean comparison and part of the goal gap is
truncation rather than policy. The crash gap is the least confounded and is 2x.

## Experiment

Cell 8x8 fill 0.3, 250k episodes, buffer 100,000, wasserstein, aggregation max,
bootstrap on, 5 seeds — i.e. identical to array 275427 except the rollout policy.

| arm | rollout policy | `cf_n_rollouts` | status |
|---|---|---|---|
| A | greedy, T=0 (target policy) | 1 | **already running**, array 275427 |
| B | ε-greedy at the *current* ε (behaviour policy) | must be > 1 | needs code |
| C | softmax T=0.5 (shipped) | 20 | on disk, n=16 at this cell |

Arm B is the only new compute. It needs the trainer's current ε plumbed into
`_build_rollout_fn`, which today closes over `temperature` only:

```python
if temperature and temperature > 0:
    action = jax.random.categorical(ak, q / temperature)
else:
    action = jnp.argmax(q)
```

becomes a third branch selecting uniform-random with probability ε, else argmax.
Jeremy's compute argument from #20 applies **here and only here**: under ε-greedy the
explore steps do not need the Q forward pass at all.

Cost: n must go back above 1 for arm B (the rollout is stochastic again), so budget it
at the shipped rate, ~8.6 h/run × 5 seeds ≈ 43 GPU-h, plus a smoke.

## Decision this feeds

Either

1. **Define π as the greedy/target policy** in Section 3, one sentence. Defensible on
   its own terms — `evaluate()` measures the greedy policy, every win rate in the paper
   is greedy, and DQN's target policy is greedy. This makes arm A the algorithm and
   `n=1` exact. Zero extra compute.
2. **Keep π = behaviour policy** and implement arm B. Literal reading of l.145, but it
   costs back the 20x that bought the coverage gain.

Arm B is what tells us whether the choice matters empirically or is only a definitional
tidy-up. If A and B land in the same place, option 1 is free and honest.
